pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java

```diff
@@ -129,6 +135,8 @@ public static void generate(PulsarService pulsar, boolean includeTopicMetrics, b
             }
             out.write(buf.array(), buf.arrayOffset(), buf.readableBytes());
         } finally {
+            //release all the metrics buffers
+            metricStreams.releaseAll();
```

Comment:
@tjiuming I noticed you had removed the `exceptionHappens`, if we don't have the check, will we release the `buf` multiple times?

From the master branch:

```
} catch (Throwable t) {
            exceptionHappens = true;
            throw t;
        } finally {
            //release all the metrics buffers
            metricStreams.releaseAll();
            //if exception happens, release buffer
            if (exceptionHappens) {
                buf.release();
            }
        }
```

And it looks like we will also cherry-pick https://github.com/apache/pulsar/pull/14453 to branch-2.9?

Reply:
@codelipenghui No, this check is for https://github.com/apache/pulsar/pull/14453, the feature didn't check-pick to 2.9

Reply:
@tjiuming will the `buf` can get a chance to be released multiple times? Since the `metricStreams.releaseAll()` will also get a chance to release the `buf`?

Reply:
@codelipenghui `buf` and `metricStreams` only released in the `finally` scope, and `metricStreams.releaseAll()` will not release `buf`

Reply:
I'm not entirely sure if this is the origin of the bug:

We allocate the byte buffer, use it for the `SimpleTextOutputStream`, and then once we finish writing to it, we return the byte buffer. We can't use `finally` as is to release the byte buffer. Only when there is an exception, need to revert the allocation of this byte buffer.
The original code is correct.

Reply:
@tjiuming Oh, I see. They are different buffers.